packages/form/addon/components/cf-field/input/files.js

```diff
@@ -0,0 +1,111 @@
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { macroCondition, isTesting } from "@embroider/macros";
+import Component from "@glimmer/component";
+import { queryManager } from "ember-apollo-client";
+import fetch from "fetch";
+
+import getFilesAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/filesanswer-info.graphql";
+
+export default class CfFieldInputFilesComponent extends Component {
+  @service intl;
+
+  @queryManager apollo;
+
+  get files() {
+    return this.args.field?.answer?.value;
+  }
+
+  @action
+  async download(fileId) {
+    if (!fileId) {
+      return;
+    }
+    const answers = await this.apollo.query(
+      {
+        query: getFilesAnswerInfoQuery,
+        variables: { id: this.args.field.answer.raw.id },
+        fetchPolicy: "network-only",
+      },
+      "node.value"
+    );
+    const { downloadUrl } =
+      answers.find((file) =>
+        macroCondition(isTesting())
+          ? file.id === fileId ||
+            atob(file.id).substring(file.__typename.length + 1) === fileId
+          : file.id === fileId
+      ) ?? {};
+    if (downloadUrl) {
+      window.open(downloadUrl, "_blank");
+    }
+  }
+
+  @action
+  async save({ target }) {
+    // store the old list of files
+    const fileList = [...(this.files || [])];
+
+    // unwrap files from FileList construct
+    const newFiles = Array.from(target.files).map((file) => ({
+      name: file.name,
+      value: file,
+    }));
+    newFiles.forEach((file) => fileList.push({ name: file.name }));
+
+    if (newFiles.length === 0) {
+      return;
+    }
+
+    // trigger save action for file list of old and new files with
+    // reduces properties to match gql format
+    const { filesValue } = await this.args.onSave(
+      fileList.map(({ name, id }) => ({ name, id }))
```

Comment:
as far as i can tell, the only property you're filtering out here is `value`. Is that really causing problems in the callsite?

Reply:
It does indeed has to have this certain form.

We defined the answer input like this in `packages/testing/addon/mirage-graphql/schema.graphql`:
```gql
input SaveDocumentFilesAnswerInput {
  question: ID!
  document: ID!
  meta: JSONString
  value: [SaveFile]
  clientMutationId: String
}
input SaveFile {
  id: String
  name: String
}
```

Reply:
![image](https://user-images.githubusercontent.com/10029904/182788084-6f74794c-7d52-4f99-ad8f-2f6e106a1450.png)

Indeed it needs this certain format since we defined it in `packages/testing/addon/mirage-graphql/schema.graphql`:
```gql
input SaveDocumentFilesAnswerInput {
  question: ID!
  document: ID!
  meta: JSONString
  value: [SaveFile]
  clientMutationId: String
}
input SaveFile {
  id: String
  name: String
}
```